src/lib.rs

```diff
@@ -448,8 +453,27 @@ impl MetadataCommand {
         self
     }
     /// Which features to include.
+    ///
+    /// Call this multiple times to specify advanced feature configurations:
+    ///
+    /// ```no_run
+    /// # use cargo_metadata::{CargoOpt, MetadataCommand};
+    /// MetadataCommand::new()
+    ///     .features(CargoOpt::NoDefaultFeatures)
+    ///     .features(CargoOpt::SomeFeatures(vec!["feat1".into(), "feat2".into()]))
+    ///     .features(CargoOpt::SomeFeatures(vec!["feat3".into()]))
+    ///     // ...
+    ///     # ;
+    /// ```
+    ///
+    /// Unlike `cargo metadata`, which disallows multiple `--all-features` or `--no-default-features`
```

Comment:
we could assert `!self.all_features` in the `AllFeatures` branch and do the same thing for `no_default_features`, too. Then the behaviour would be the same.

Reply:
Sounds good. We'll replicate `cargo metadata` behaviors, and `panic!()` if calling `features(AllFeatures)` or `features(NoDefaultFeatures)` more than once.